### ISSUE TITLE
Downgrade gothemis compatibility

### ISSUE THREAD

**Issue by vixentael** [cossacklabs/themis#253]
It seems that gothemis uses syntax incompatible with old Go versions (TODO: check exact version, but it's definitely below 1.2).

We should re-write this line to smth more compatible
https://github.com/cossacklabs/themis/blob/master/gothemis/session/session.go#L93

As example we used this code in Hermes:

```
//CArrayToSlice return buffer as byte slice with len/cap equals to bufferLength
func CArrayToSlice(buffer unsafe.Pointer, bufferLength int) []byte {
	sliceHeader := reflect.SliceHeader{uintptr(buffer), int(bufferLength), bufferLength}
	data := *(*[]byte)(unsafe.Pointer(&sliceHeader))
	return data
}
```
(from https://github.com/cossacklabs/hermes-core/blob/master/gohermes/utils.go)

Although this code is 100% valid, but it would be nice to support previous versions.

**Comment by vixentael** [comment]
> Self-check:
> Internal buildbot builds that are supposed to be fixed after this fix:
>
> - themis-test-ubuntu_precise-x86_64
> - themis-test-ubuntu_precise-i386
> - themis-test-debian_wheezy-i386

**Comment by vixentael** [comment]
closed by #254
